app/lib/account/backend.dart

```diff
@@ -87,31 +87,71 @@ UserSessionData? get userSessionData =>
 /// a new one. When the authenticated email of the user changes, the email
 /// field will be updated to the latest one.
 Future<AuthenticatedUser> requireAuthenticatedWebUser() async {
-  return await _requireAuthenticatedUser(
-      expectedAudience: activeConfiguration.pubSiteAudience);
+  final agent = await _requireAuthenticatedAgent(
+      expectedUserAudience: activeConfiguration.pubSiteAudience);
+  if (agent is! AuthenticatedUser) {
+    throw AuthenticationException.failed();
+  }
+  return agent;
 }
 
-Future<AuthenticatedUser> _requireAuthenticatedUser(
-    {String? expectedAudience}) async {
+/// Require that the incoming request is authorized by an administrator with
+/// the given [permission].
+///
+/// Throws [AuthorizationException] if it doesn't have the permission.
+Future<AuthenticatedUser> requireAuthenticatedAdmin(
+    AdminPermission permission) async {
+  final agent = await _requireAuthenticatedAgent(
+      expectedUserAudience: activeConfiguration.adminAudience);
+  if (agent is! AuthenticatedUser) {
+    throw AuthenticationException.failed();
+  }
```

Comment:
I think this will break admin.
I think admin will now be `AuthenticatedGcpServiceAccount` objects.

Can't we just return a `AuthenticatedGcpServiceAccount`?

We can also do this in a follow up PR -- but right now I think admin is broken.

Reply:
Ah, true, in my head we did change the admin audience (#6136), but it was discarded.